### PR TITLE
Add form checks

### PR DESCRIPTION
**Pull request by stevenvegt**
Add min date of today for transferDate
Add max date of today for patient date of birth
Add empty checks for patient and transfer fields

### CODE REVIEW

File: web/src/ehr/transfer/NewTransfer.vue [hunk 1]
```diff
@@ -34,10 +44,56 @@ export default {
...
+        if (patientProblem.problem.name === "") {
+          this.formErrors.push("Each problem must have a description.")
+          return false
+        }
+        if (patientProblem.interventions.length == 0) {
+          this.formErrors.push("Each problem must have at least 1 intervention.")
+        }
+      })
+      return this.formErrors.length == 0
+    },
```
Comment (dmeijboom) [inline web/src/ehr/transfer/NewTransfer.vue:89]: Can this be de-spaghettified? It's really hard to read. Maybe something like this (untested):

```

      this.transfer.carePlan.patientProblems.forEach((patientProblem, i) => {
        // skip problems placeholder
        if (this.transfer.carePlan.patientProblems.length - 1 === i &&
            !patientProblem.problem.name &&
            patientProblem.interventions.filter(intervention => !!intervention.comment).length === 0) {
          return;
        }

        // skip the intervention placeholder
        patientProblem.interventions.slice(0, patientProblem.interventions.length - 1).forEach(intervention => {
          if (!intervention.comment) {
            this.formErrors.push("Each intervention must have a comment.")
          }
        });

        if (!patientProblem.problem.name) {
          this.formErrors.push("Each problem must have a description.")
        }
        
        if (patientProblem.interventions.length === 0) {
          this.formErrors.push("Each problem must have at least 1 intervention.")
        }
      });
```

**Review by dmeijboom — CHANGES_REQUESTED** [review]
It would also be nice if all forms use `@submit.prevent` for the form handler and the buttons are just regular `type="submit"` buttons. That way pressing enter on the last field also works for example.

File: web/src/ehr/transfer/NewTransfer.vue (hunk 1 above)
Comment (stevenvegt) [reply web/src/ehr/transfer/NewTransfer.vue:92]: This gives me headaches. Is it really cleaner?

Comment (dmeijboom) [reply web/src/ehr/transfer/NewTransfer.vue:92]: Going back to the original source the issue I have is that it takes a while to understand what the code does. At also needs 4 comments to make sense. At the same time it's a simple algorithm but it contains a lot of exceptions (error when empty but not if it's the last or first section) which makes it hard to keep the code simple.

What I tried to do is to get rid of things like this:

```js
if ((i > 0 || j != 0) && j == patientProblem.interventions.length - 1) {
```

An alternative could be to get rid of the placeholders before validation (basically copy the data, delete the placeholders then validate). That would make it a lot more straightforward and we can drop checking for empty interventions in the backend. I don't want to make this a big deal as it's a simple validation used by a single component but I would like the code to be a little bit cleaner.

What do you think?

**Review by dmeijboom — APPROVED** [review]
LGTM 👍

**Pull request merged**